src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp

```diff
@@ -6044,6 +6044,110 @@ GUI_TEST_CLASS_DEFINITION(test_6860) {
+    GTUtilsWorkflowDesigner::clickParameter(os, "Accept flag");
+    GTUtilsWorkflowDesigner::clickParameter(os, "Accept flag");
+    checkboxValue = GTComboBox::getCurrentText(os, GTWidget::findExactWidget<QComboBox *>(os, "mainWidget"));
+    CHECK_SET_ERR(checkboxValue == "Mate strand", QString("Accept flag value (1): expected \"Mate strand\", current: \"%1\"").arg(checkboxValue))
+
+    // To successfully complete the test
+    GTWidget::click(os, GTWidget::findWidget(os, "sceneView"));
+}
+
+GUI_TEST_CLASS_DEFINITION(test_6862_1) {
+    class CheckWizardWidget : public CustomScenario {
+    public:
+        void run(GUITestOpStatus &os) override {
+            for (int i = 0; i < 5; i++) {
+                GTUtilsWizard::clickButton(os, GTUtilsWizard::Next);
+            }
+            QString value = GTComboBox::getCurrentText(os, GTWidget::findExactWidget<QComboBox *>(os, "Motif database widget"));
+            CHECK_SET_ERR(value == "hpdi.xml", QString("Motif database value (2): expected \"hpdi.xml\", current: \"%1\"").arg(value))
+
+            GTComboBox::checkValues(os, GTWidget::findExactWidget<QComboBox *>(os, "Motif database widget"), QStringList("cistrome.xml"));
+            GTKeyboardDriver::keyClick(Qt::Key_Enter);
+
+            GTUtilsWizard::clickButton(os, GTUtilsWizard::Next);
+            GTUtilsWizard::clickButton(os, GTUtilsWizard::Apply);
+        }
+    };
+
+    //      Make sure the changes did not destroy good behavior in
+    //              . Clicking elsewhere after switching in a checkbox
+    //              . Wizard
+    //              . ComboBoxWithChecksWidget derived class
+    // Open data/workflow_samples/NGS/cistrome/chip_seq.uwl
+    // Select Collect Motifs with SeqPos element
+    // Change Motif database parameter from "cistrome.xml" to "hpdi.xml"
+    // Click on the name of the Motif database parameter
+    //     Expected state: Motif database value is "hpdi.xml"
+    // Click on empty place on the scene
+    // Call schema wizard
+    // Click Next 5 times
+    //     Expected state: Motif database value is "hpdi.xml"
+    // Set "cistrome.xml" as the Motif database parameter
+    // Click Next and Apply
+    //     Expected state: Motif database value is "cistrome.xml"
+
+    StartupDialogFiller *filler = new StartupDialogFiller(os);
+    GTUtilsDialog::waitForDialogWhichMayRunOrNot(os, filler);
+    GTFileDialog::openFile(os, dataDir + "workflow_samples/NGS/cistrome/chip_seq.uwl");
+
+    GTUtilsWorkflowDesigner::click(os, "Collect Motifs with SeqPos");
+    GTUtilsWorkflowDesigner::setParameter(os, "Motif database", "hpdi.xml", GTUtilsWorkflowDesigner::ComboChecks);
+
+    // Click on the name of the Motif database parameter
+    QTableView *table = GTUtilsWorkflowDesigner::getParametersTable(os);
+    QAbstractItemModel *parametersModel = table->model();
+    CHECK_SET_ERR(parametersModel != nullptr, "ComboBox model is nullptr")
+    table->scrollTo(parametersModel->index(4, 0));
```

Comment:
Tests are run in a separate thread. Do not use QT widgets API from a non-main thread.

Reply:
`GTUtilsWorkflowDesigner::setParameter` is implemented just like this one. How to do it right?

Reply:
Okay, I have an idea

Reply:
You are right we have some API that works incorrectly and I'm trying to migrate it to work from the main thread when it comes to QT GUI state.

The migration priority is:
1) Migrate all places that update GUI state.
2) Migrate all places that read GUI state.

Until this long term task is finished the best we can do is not to introduce new usages, at least for the 1)